Issue building nodejs-16_x.x86_64-darwin

### Describe the bug
`nodejs-16_x.x86_64-darwin` does not build. See [Hydra](https://hydra.nixos.org/build/161094289#tabs-summary).

### Steps To Reproduce
Steps to reproduce the behavior:
1. Try to build/install `nodejs-16_x`
2. See the build fail with:

```
error: builder for '/nix/store/8d9qf5mvbspmjwn440why63j03sp7xqy-nodejs-16.13.1.drv' failed with exit code 2;
       last 10 log lines:
       > /nix/store/2dlfw9s2ypb0s245g3yrq29p6rvlwsky-Libsystem-1238.60.2/include/CommonCrypto/CommonRandom.h:35:9: error: unknown type name 'CCCryptorStatus'
       > typedef CCCryptorStatus CCRNGStatus;
       >         ^
       > ../deps/openssl/openssl/crypto/rand/rand_unix.c:385:47: error: use of undeclared identifier 'kCCSuccess'
       >     if (CCRandomGenerateBytes(buf, buflen) == kCCSuccess)
       >                                               ^
       > 2 errors generated.
       > make[1]: *** [deps/openssl/openssl.target.mk:886: /private/tmp/nix-build-nodejs-16.13.1.drv-0/node-v16.13.1/out/Release/obj.target/openssl/deps/openssl/openssl/crypto/rand/rand_unix.o] Error 1
       > rm e1da146381dd9f1a7ed9a4ea14428d0a896ac4ce.intermediate
       > make: *** [Makefile:110: node] Error 2
       For full logs, run 'nix log /nix/store/8d9qf5mvbspmjwn440why63j03sp7xqy-nodejs-16.13.1.drv'.
```

### Expected behavior
It should build successfully.

### Notify maintainers
@marsam @gilligan @cko @goibhniu

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.1.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4`
 - channels(daniel): `"home-manager"`
 - channels(root): `"nixpkgs-21.11pre330734.5cb226a06c4"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/development/web/nodejs/v16.nix
+++ b/pkgs/development/web/nodejs/v16.nix
@@ -1,4 +1,4 @@
-{ callPackage, openssl, python3, enableNpm ? true }:
+{ callPackage, fetchpatch, openssl, python3, enableNpm ? true }:
 
 let
   buildNodejs = callPackage ./nodejs.nix {
@@ -10,5 +10,13 @@ in
     inherit enableNpm;
     version = "16.13.1";
     sha256 = "1bb3rjb2xxwn6f4grjsa7m1pycp0ad7y6vz7v2d7kbsysx7h08sc";
-    patches = [ ./disable-darwin-v8-system-instrumentation.patch ];
+    patches = [
+      ./disable-darwin-v8-system-instrumentation.patch
+      # Fixes node incorrectly building vendored OpenSSL when we want system OpenSSL.
+      # https://github.com/nodejs/node/pull/40965
+      (fetchpatch {
+        url = "https://github.com/nodejs/node/commit/65119a89586b94b0dd46b45f6d315c9d9f4c9261.patch";
+        sha256 = "sha256-dihKYEdK68sQIsnfTRambJ2oZr0htROVbNZlFzSAL+I=";
+      })
+    ];
   }

